src/storage/txn/commands/prewrite.rs

```diff
@@ -546,11 +548,14 @@ trait PrewriteKind {
     }
 }
 
+/// Information needed for the optimistic variant of `PreWriteKind`.
 struct Optimistic {
     skip_constraint_check: bool,
 }
 
+/// The optimistic variant of `PreWriteKind`.
 impl PrewriteKind for Optimistic {
+    // see the `MutationLock` trait implementation for `Mutation`
```

Comment:
ditto

Reply:
I agree. I think after documenting both the Trait and the Struct, we can skip the doc for the impl.